tests/test_mypy.py

```diff
@@ -0,0 +1,93 @@
+import shutil
+import textwrap
+import os
+import unittest
+
+mypy_installed = True
+try:
+    import mypy.api
+except ImportError:
+    # mypy not installed on pypy (see setup.py)
+    mypy_installed = False
+
+HERE = os.path.dirname(__file__)
+TEST_OUTPUT_DIR = os.path.join(HERE, "test-output")
+MYPY_INI = """\
+[mypy]
+follow_imports = silent
+plugins = marshmallow_dataclass.mypy
+"""
+
+
+@unittest.skipUnless(mypy_installed, "mypy required")
+class TestMypyPlugin(unittest.TestCase):
+    maxDiff = None
+
+    def setUp(self):
+        """
+        Prepare a clean test directory at tests/test-output/test_mypy-{testname}.
+        Also cd into it for the duration of the test to get simple filenames in mypy output.
+        """
+        testname = self.id().split(".")[-1]
+        self.testdir = os.path.join(TEST_OUTPUT_DIR, f"test_mypy-{testname}")
```

Comment:
Hah, yeah I must admit I was bit surprised that this was using `unittest`. Note that there's also a [pytest plugin](https://pypi.org/project/pytest-mypy-plugins/) to easily test mypy plugins, which would effectively remove the need for all the code in `setup`, `teardown` and `assert_mypy_output`.
Do you want me to leave a `TODO` somewhere, as a reminder to switch to (or at least consider) `pytest-mypy-plugins` if the project moves to `pytest`?

Also note that [pytest supports unittest](https://docs.pytest.org/en/latest/unittest.html), so you could use the pytest runner without necessarily "rewriting" the existing tests.

Reply:
I argued against pytest in the past because I felt like it wasn't providing anything we needed for this project. But if there is a plugin that allows testing the mypy plugin easily, then I have nothing against switching to pytest and using it. It would definitely make the transition worth it.

Reply:
Argh. Deleted previous comment by accident. Copied below for reference.

Hah, yeah I must admit I was bit surprised that this was using unittest. Note that there's also a pytest plugin to easily test mypy plugins, which would effectively remove the need for all the code in setup, teardown and assert_mypy_output.
Do you want me to leave a TODO somewhere, as a reminder to switch to (or at least consider) pytest-mypy-plugins if the project moves to pytest?

Also note that pytest supports unittest, so you could use the pytest runner without necessarily "rewriting" the existing tests.

Reply:
@lovasoa Cool. How do you want to proceed re: existing tests and re: this PR?

Reply:
Since switching to pytest doesn't require rewriting any of the existing tests, I think we can do it in this PR.

Reply:
Alright! Will work on that later today.